### ISSUE TITLE
An addon that allows you to see the Instructions/N&C of a project from full screen

### ISSUE THREAD

**Issue by DNin01** [ScratchAddons/ScratchAddons#4933]
Let's say you just _gotta_ use full screen mode for a project, whether it's to see more easily, to click precisely, or to suit your preference. Well, now let's say you need some help from the Instructions... **but you have to exit full screen mode to see them**. Maybe you end up doing it multiple times. Not exactly fun. That's where this new idea comes in.

My suggestion is to make an addon that allows you to see the project description areas _while in full screen mode_. **With a click of a button, the Instructions and Notes and Credits sections would show up to the right.**

**Comment by Cool-Scratcher** [comment]
Yeah okay but where would this fit? I'm assuming you mean a button to show the instructions/n&c. In that case, is this another top bar addon? That location is so overused 😢

**Comment by DNin01** [comment]
> Yeah okay but where would this fit? I'm assuming you mean a button to show the instructions/n&c. In that case, is this another top bar addon? That location is so overused 😢 

I was also thinking of a "handle" with an arrow on it at the right edge of the full screen page. This would show and hide those sections.

**Comment by BroJac5246** [comment]
> Yeah okay but where would this fit? I'm assuming you mean a button to show the instructions/n&c. In that case, is this another top bar addon? That location is so overused 😢

If so, it wouldn't be that bad, as many of the top bar addons don't appear in full screen, where you _also_ have more room.


> I was also thinking of a "handle" with an arrow on it at the right edge of the full screen page. This would be the button show and hide those sections.

I think that makes the most sense

Would this addon appear while in full screen in the project editor, or only while in full screen on a project page?

**Comment by DNin01** [comment]
> Would this addon appear while in full screen in the project editor, or only while in full screen on a project page?

Probably just for full screen on project pages - that's the only place the Instructions and N&C are seen, and you probably won't need them while you're editing.

**Comment by mxmou** [comment]
> > Would this addon appear while in full screen in the project editor, or only while in full screen on a project page?
> 
> Probably just for full screen on project pages - that's the only place the Instructions and N&C are seen, and you probably won't need them while you're editing.

I think we should avoid creating differences between project page full screen and editor full screen.

**Comment by BroJac5246** [comment]
> > > Would this addon appear while in full screen in the project editor, or only while in full screen on a project page?
> > 
> > 
> > Probably just for full screen on project pages - that's the only place the Instructions and N&C are seen, and you probably won't need them while you're editing.
> 
> I think we should avoid creating differences between project page full screen and editor full screen.

I can see that... but we have to acknowledge that they are different. As Cool-Scratcher pointed out, that area can be overused- and as we've seen demonstrated in numerous suggestions, users don't like that. I agree with DNin01 that we should have this addon only appear on the project page. If it's enabled in the editor, it just wastes valuable space.

Basically, consistency is important, but ultimately it's about the user experience. Users tend to like consistency, which is why it is preferred, however, in this case, it simply degrades the user experience by unnecessarily wasting space.

**Comment by WorldLanguages** [comment]
Exiting full screen to read the instructions doesn't seem like a real inconvenience to me

**Comment by mxmou** [comment]
> > I think we should avoid creating differences between project page full screen and editor full screen.
> 
> I can see that... but we have to acknowledge that they are different. As Cool-Scratcher pointed out, that area can be overused- and as we've seen demonstrated in numerous suggestions, users don't like that. I agree with DNin01 that we should have this addon only appear on the project page. If it's enabled in the editor, it just wastes valuable space.
> 
> Basically, consistency is important, but ultimately it's about the user experience. Users tend to like consistency, which is why it is preferred, however, in this case, it simply degrades the user experience by unnecessarily wasting space.

There is enough space in the full screen header. Opening full screen in the editor and reloading switches to the project page full screen view and it would be confusing if that caused an additional button to appear.

**Comment by BroJac5246** [comment]
> Exiting full screen to read the instructions doesn't seem like a real inconvenience to me

That doesn't mean it's very _convenient_

Anyways, why has no one thought of this, but:
_We could just add toggles to let the user choose where it appears._
That would solve this whole problem, and it's a way better solution than what any of us have suggested

**Comment by mxmou** [comment]
> Anyways, why has no one thought of this, but: _We could just add toggles to let the user choose where it appears._ That would solve this whole problem, and it's a way better solution than what any of us have suggested

Letting the user choose is usually not a good way to solve problems.

**Comment by BroJac5246** [comment]
> > Anyways, why has no one thought of this, but: _We could just add toggles to let the user choose where it appears._ That would solve this whole problem, and it's a way better solution than what any of us have suggested
> 
> Letting the user choose is usually not a good way to solve problems.

We don't have a problem- just a disagreement about where this addon should run. Let the user decide, and they get a more customizable experience.

As long as we've tested that it works in both locations, we won't have a problem.

**Comment by Secret-chest** [comment]
Put it on the right side and maje it collapsible. On 16:9 screens there is enough space there...

**Comment by WorldLanguages** [comment]
> We don't have a problem- just a disagreement about where this addon should run.

Adding additional, unrelated UI elements to the full screen page can create problems for other addons, that will need to take those into account. This would also potentially break the assumption that project page full screen and editor full screen are (almost) the same.

**Comment by BroJac5246** [comment]
> > We don't have a problem- just a disagreement about where this addon should run.
> 
> Adding additional, unrelated UI elements to the full screen page can create problems for other addons, that will need to take those into account. This would also potentially break the assumption that project page full screen and editor full screen are (almost) the same.

That's why we need to make it clear that it _will_ be different if you configure it to be that way.
